core/src/main/java/hudson/slaves/JNLPLauncher.java

```diff
@@ -221,4 +221,18 @@ public boolean filterType(@Nonnull Class<?> contextClass, @Nonnull Descriptor de
         }
     }
 
+    /**
+     * Returns true if Java Web Start button should be displayed.
+     * Java Web Start is only supported when the Jenkins server is
+     * running with Java 8.  Earlier Java versions are not supported by Jenkins.
+     * Later Java versions do not support Java Web Start.
+     *
+     * This flag is checked in {@code config.jelly} before displaying the
+     * Java Web Start button.
+     * @return {@code true} if Java Web Start button should be displayed.
+     * @since FIXME
+     */
+    public boolean isJavaWebStartSupported() {
+        return System.getProperty("java.version", "1.8").startsWith("1.8");
```

Comment:
I was looking for https://docs.oracle.com/javase/10/docs/api/java/lang/Runtime.Version.html but it's only available since Java 9, sadly. So I guess this `startWith` is really all we can do here..

Reply:
That's what I saw as well.  I would have preferred a better check, but...

Reply:
This is a hack we usually do. Probably we need to move these checks to a utility class